lib/Runtime/Library/AtomicsObject.cpp

```diff
@@ -0,0 +1,270 @@
+
+    Var AtomicsObject::EntryLoad(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(2, "Atomics.load");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+        return typedArrayBase->TypedLoad(accessIndex);
+    }
+
+    Var AtomicsObject::EntryOr(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(3, "Atomics.or");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+        return typedArrayBase->TypedOr(accessIndex, args[3]);
+    }
+
+    Var AtomicsObject::EntryStore(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(3, "Atomics.store");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+        return typedArrayBase->TypedStore(accessIndex, args[3]);
+    }
+
+    Var AtomicsObject::EntrySub(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(3, "Atomics.sub");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+        return typedArrayBase->TypedSub(accessIndex, args[3]);
+    }
+
+    Var AtomicsObject::EntryWait(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(3, "Atomics.wait");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext, true /*onlyInt32*/);
+
+        Var value = args[3];
+        Var timeoutVar = args[4];
+        uint32 timeout = INFINITE;
+
+        if (args.Info.Count > 4 && !JavascriptOperators::IsUndefinedObject(timeoutVar))
+        {
+            double t =JavascriptConversion::ToNumber(timeoutVar, scriptContext);
+            if (!(NumberUtilities::IsNan(t) || JavascriptNumber::IsPosInf(t)))
+            {
+                int32 t1 = JavascriptConversion::ToInt32(t);
+                timeout = (uint32)max(0, t1);
+            }
+        }
+
+        if (!AgentOfBuffer::AgentCanSuspend(scriptContext))
+        {
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_CannotSuspendBuffer);
+        }
+
+        Assert(typedArrayBase->GetBytesPerElement() == 4);
+        uint32 bufferIndex = (accessIndex * 4) + typedArrayBase->GetByteOffset();
+        Assert(bufferIndex < typedArrayBase->GetArrayBuffer()->GetByteLength());
+        SharedArrayBuffer *sharedArrayBuffer = typedArrayBase->GetArrayBuffer()->GetAsSharedArrayBuffer();
+        WaiterList *waiterList = sharedArrayBuffer->GetWaiterList(bufferIndex);
+
+        int32 v = JavascriptConversion::ToInt32(value, scriptContext);
```

Comment:
>int32 v = JavascriptConversion::ToInt32(value, scriptContext); [](start = 8, length = 62)

Move these before timeoutVar check as spec have this (step 3) before the timeout check (step 4) and ToInt32 can potentially throw